.github/workflows/build.yaml

```diff
@@ -241,8 +241,9 @@ jobs:
             output/images/haos_ova*.qcow2.xz
 
   bump_version:
-    name: Bump dev channel version
+    name: Bump ${{ needs.prepare.outputs.channel }} channel version
     if: ${{ github.repository == 'home-assistant/operating-system' }}
+    environment: "${{ github.event_name == 'release' && needs.prepare.outputs.channel || '' }}"
```

Comment:
Whats the reason behind providing a `""` environment here, instead of just using `needs.prepare.outputs.channel` like on core <https://github.com/home-assistant/core/blob/dev/.github/workflows/builder.yml#L53>

Reply:
Originally the idea was to have a single pre-defined environment with protections created before the first run, and not to create any other environments in case they don't exist yet (which is what GH does - if you specify a non-existing `environment`, it creates it). But after changing it to channel name per Stefan's suggestion, creating a third one for dev wouldn't do any harm now...

Reply:
Hm, I was under the assumption that an environment always has some protection rule, but it seems that can be bypassed completely. I agree we should just use channel directly, and create a `dev` environment without protection rules. :+1: 

There is another problem in our current build pipeline: Once we release a final release, beta should receive a bump as well (see https://github.com/home-assistant/core/blob/dev/.github/workflows/builder.yml#L309-L316). 

@sairon can you add this to the pipeline as well?